docs/Routes.md

```diff
@@ -168,7 +171,25 @@ fastify.get('/', options, async function (request, reply) {
   reply.send(processed)
 })
 ```
-**Warning:** If you use `return` and `reply.send` at the same time, the first one that happens takes precedence, the second value will be discarded, a *warn* log will also be emitted if the value is not `undefined`.
+**Warning:**
+* If you use `return` and `reply.send` at the same time, the first one that happens takes precedence, the second value will be discarded, a *warn* log will also be emitted because you tried to send a response twice.
+* You can't return `undefined`. For more details read [promise-resolution](#promise-resolution).
+
+<a name="promise-resolution"></a>
+### Promise resolution
+
+If you use `async/await` or return a promise inside your handler you should be aware of a special behaviour which is necessary to support the callback and promise control-flow. If you resolve the handler's promise with `undefined`, it will be ignored causing the request to hang and an *error* log to be emitted.
```

Comment:
Too many "you"

If you resolve the handler's promise
If the handler's promise is resolved

Reply:
PR? :)